components/widgets/menu/menu_items/menu_item.tsx

```diff
@@ -12,16 +14,61 @@ export default function menuItem(Component: React.ComponentType<any>) {
         id?: string;
         icon?: React.ReactNode;
         text?: React.ReactNode;
+        index?: number;
+        size?: number;
+    }
+
+    type State = {
+        currentFocus: number;
+        size: number;
+        isCompUnmounted: boolean;
+        isListenerAdded: boolean;
     }
-    class MenuItem extends React.PureComponent<Props & React.ComponentProps<typeof Component>> {
+    class MenuItem extends React.PureComponent<Props & React.ComponentProps<typeof Component>, State> {
+        constructor(props: Props) {
+            super(props);
+
+            this.state = {
+                currentFocus: 0,
+                size: this.props.size,
+                isCompUnmounted: true,
+                isListenerAdded: true,
+            };
+        }
+
+        setRoveFocus = () => {
```

Comment:
If we end up using this function please put a comment above with credit to the original article https://dev.to/rafi993/roving-focus-in-react-with-custom-hooks-1ln

Reply:
This appears to break tabbing and arrow navigating in all other parts of the app that use the menu component. Once I open a place that uses this component, tabbing completely breaks everywhere apart from the add_channel_dropdown component.

Reply:
Also shift+tab no longer works